src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -0,0 +1,231 @@
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_hidden_nonlinearity=torch.tanh,
+                 std_hidden_w_init=nn.init.xavier_uniform_,
+                 std_hidden_b_init=nn.init.zeros_,
+                 std_output_nonlinearity=None,
+                 std_output_w_init=nn.init.xavier_uniform_,
+                 std_parameterization='exp',
+                 layer_normalization=False):
+        super(GaussianMLPModule, self).__init__()
+
+        self._input_dim = input_dim
+        self._hidden_sizes = hidden_sizes
+        self._action_dim = output_dim
+        self._learn_std = learn_std
+        self._adaptive_std = adaptive_std
+        self._std_share_network = std_share_network
+        self._std_hidden_sizes = std_hidden_sizes
+        self._min_std = min_std
+        self._max_std = max_std
+        self._std_hidden_nonlinearity = std_hidden_nonlinearity
+        self._std_hidden_w_init = std_hidden_w_init
+        self._std_hidden_b_init = std_hidden_b_init
+        self._std_output_nonlinearity = std_output_nonlinearity
+        self._std_output_w_init = std_output_w_init
+        self._std_parameterization = std_parameterization
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._hidden_w_init = hidden_w_init
+        self._hidden_b_init = hidden_b_init
+        self._output_nonlinearity = output_nonlinearity
+        self._output_w_init = output_w_init
+        self._output_b_init = output_b_init
+        self._layer_normalization = layer_normalization
+
+        self._init_std_param = torch.Tensor([init_std])
+        if self._learn_std:
+            self._init_std = torch.nn.Parameter(self._init_std_param)
+
+        self._min_std_param, self._max_std_param = None, None
+        if min_std:
+            min_std = torch.tensor(min_std, dtype=torch.float)
+        if max_std:
+            max_std = torch.tensor(max_std, dtype=torch.float)
+
+        if self._std_parameterization == 'exp':
+            self._init_std_param = self._init_std_param.log()
+            self._min_std_param = min_std.log() if min_std else None
+            self._max_std_param = max_std.log() if max_std else None
+
+        elif self._std_parameterization == 'softplus':
+            self._init_std_param = self._init_std_param.exp().add(-1).log().log()
+            self._min_std_param = min_std.exp().add(-1).log().log() if min_std else None
+            self._max_std_param = max_std.exp().add(-1).log().log() if max_std else None
+
+        else:
+            raise NotImplementedError
+
+        if std_share_network:
+            self.shared_mean_log_std_network = self._get_shared_mean_std_network()
```

Comment:
is there a reason that this attribute is public?

Reply:
no, fixed.